lib/tapioca/dsl/helpers/active_record_column_type_helper.rb

```diff
@@ -119,6 +121,23 @@ def enum_setter_type(column_type)
             "T.any(::String, ::Symbol)"
           end
         end
+
+        sig { params(column_type: ActiveRecord::Type::Serialized).returns(String) }
+        def serialized_column_type(column_type)
+          case column_type.coder
+          when ActiveRecord::Coders::YAMLColumn
+            case column_type.coder.object_class
+            when Array.singleton_class
+              "T::Array[T.untyped]"
+            when Hash.singleton_class
+              "T::Hash[T.untyped, T.untyped]"
+            else
+              "T.untyped"
+            end
+          else
+            "T.untyped"
+          end
```

Comment:
From what I understand serialization formats only give guarantees for `Array` and `Hash` types, and everything else should just be `T.untyped`:
```
serialize attr_name [, class_name_or_coder]

                     |                           |  database storage   |
class_name_or_coder  | attribute read/write type | serialized | NULL   |
---------------------+---------------------------+------------+--------+
  <not given>        | any value that supports   |    YAML    |        |
                     |   .to_yaml                |            |        |
                     |                           |            |        |
Array                | Array **                  |    YAML    |  []    |
                     |                           |            |        |
Hash                 | Hash **                   |    YAML    |  {}    |
                     |                           |            |        |
JSON                 | any value that supports   |    JSON    |        |
                     |   .to_json                |            |        |
                     |                           |            |        |
<custom coder class> | any value supported by    |   custom   | custom |
                     | the custom coder class    |            |        |
```

Representing `JSON` as a `Hash` type or generic YAML as an `Object` type would not be correct, since they serialize anything that responds to `to_json` and `to_yaml` respectively.

Reply:
`class_name_or_coder` not given also checks if the object inherits from `Object`, so `BasicObject` isn't allowed. The signature of `T.nilable(Object)` seems correct for that case.

Reply:
If we type that return value as `T.nilable(Object)`, it will be correct but not very ergonomic:
```ruby
extend T::Sig

class Foo
  def foo
  end
end

sig { returns(Object) }
def serialized_field
  Foo.new
end

f = serialized_field
f.foo # Error: Method foo does not exist on Object
```
Since, yes, the return value technically "is a" `Object` but most of the time you want to access specific functionality on it.

Because we don't know what type was stored in the DB field, I'd rather we type it as `T.untyped`.

Reply:
yeah, good point. Agree.